### PR TITLE
#184759900 implemement google auth

### PR DESCRIPTION
**Pull request by ndatumuremyi**
### What does this PR do?
This PR implements Google authentication

### Description of Tasks to be completed?
 - Users should be able to log in via Google. 

### How should this be manually tested?
1. start [Vikings backend](https://github.com/atlp-rwanda/vikings-ec-bn.git) at port 5000 with GOOGLE_CALLBACK_URL=http://localhost:8000/redirect-google in your env
2. Clone [repo](https://github.com/atlp-rwanda/vikings-ec-fe.git)
3. Change into the directory `vikings-ec-fe`
4. Checkout to the branch: `feature/implement-google-auth-184759900`
5. Run the command `npm install` to install all packages
6. Run the command `npm run start`
7. go to `/auth`
8. click on the `google` icon

#### Screenshot
gooleLogin

### Figma

loginWithGoogleFigma

### What are the relevant pivotal tracker stories?
[#184759900](https://www.pivotaltracker.com/story/show/184759900)

### What are the relevant pivotal tracker stories?
[#184759900](https://www.pivotaltracker.com/story/show/184759900)

### CODE REVIEW

**Review by Abdden — APPROVED** [review]

**Review by irakozetony — APPROVED** [review]

**Review by Aline-Uwera — APPROVED** [review]

File: src/pages/HomePage.jsx [hunk 1]
```diff
@@ -6,15 +6,20 @@ import getMessage from '../features/actions/welcomeAction';
 function HomePage() {
   const dispatch = useDispatch();
   const { data } = useSelector((state) => state.message);
+  const { data: userData } = useSelector((state) => state.auth);
   useEffect(() => {
+    console.log('backend ', process.env.REACT_APP_BASE_URL);
```
Comment (ncutixavier) [inline src/pages/HomePage.jsx]: Remove all logs

File: src/features/actions/authActions.js
Comment (ncutixavier) [inline src/features/actions/authActions.js:1]: Rename this file to be specific for google

File: src/features/auth/authSlice.js
Comment (ncutixavier) [inline src/features/auth/authSlice.js:1]: Rename this file

File: src/store.js [hunk 2]
```diff
@@ -10,6 +11,7 @@ if (process.env.NODE_ENV === 'development') {
 const store = configureStore({
   reducer: {
     message: welcomeSlice,
+    auth: authSliceReducers,
```
Comment (ncutixavier) [inline src/store.js]: Change this key to be specific for google auth

File: src/utils/Keys.js
Comment (ncutixavier) [inline src/utils/Keys.js:1]: This is not needed

File: src/features/api/customAxios.js [hunk 3]
```diff
@@ -1,36 +1,37 @@
 import axios from 'axios';
+import Keys from '../../utils/Keys';
 
 const customAxios = axios.create({
-  baseURL: `https://vikings-ec-bn-mbhd.onrender.com/api/v1`,
+  baseURL: `${Keys.BASE_URL}`,
```
Comment (ncutixavier) [inline src/features/api/customAxios.js]: Just get the base URL from  `.env`

**Review by ncutixavier — CHANGES_REQUESTED** [review]
In the tests folder, The folder and file naming is not meaningful. For example, 
- there are folders called `mocs`, `redux.test`, Please change them to be specific
- Some files need to be renamed

**Review by ncutixavier — COMMENTED** [review]
Change wherever you used `google` to `googleAuth`

**Review by Aline-Uwera — APPROVED** [review]

File: environments/.env [hunk 4]
```diff
@@ -0,0 +1 @@
+REACT_APP_BASE_URL=https://vikings-ec-bn-mbhd.onrender.com/api/v1
```
Comment (ncutixavier) [inline environments/.env]: Please don't push env variables.

File: src/components/GoogleAuthButton.jsx [hunk 5]
```diff
@@ -0,0 +1,13 @@
+import React from 'react';
+import { Link } from 'react-router-dom';
+import GoogleIcon from '../../public/images/icons/google_icon.svg';
+
+const GoogleAuthButton = () =>
+  // const message = 'button';
```
Comment (ncutixavier) [inline src/components/GoogleAuthButton.jsx]: Remove unnecessary comments

File: src/features/auth/googleSlice.js [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+  initialState,
+  extraReducers: {
+    [googleRedirect.pending]: (state) => {
+      state.isLoading = true;
+    },
+    [googleRedirect.fulfilled]: (state, { payload }) => {
+      state.isLoading = false;
+      state.data = payload;
+      // eslint-disable-next-line no-undef
+      localStorage.setItem('token', payload.token);
```
Comment (ncutixavier) [inline src/features/auth/googleSlice.js]: Put this in actions

**Review by ncutixavier — CHANGES_REQUESTED** [review]
After signup with Google, Thisam not being redirected to any page, I've received this JSON
image

**Pull request merged**